codespan-reporting/tests/snapshots/term__multifile__rich_no_color.snap

```diff
@@ -22,16 +22,15 @@ warning: unused parameter pattern: `n₂`
 
 error[E0001]: unexpected type in application of `_+_`
 
-   ┌── Test.fun:4:11 ───
-   │
- 4 │ _ = 123 + "hello"
-   │           ^^^^^^^ expected `Nat`, found `String`
-   │
-
     ┌── Data/Nat.fun:11:1 ───
     │
  11 │ _+_ : Nat → Nat → Nat
     │ --------------------- based on the definition of `_+_`
     │
+    ┌── Test.fun:4:11 ───
+    │
+  4 │ _ = 123 + "hello"
+    │           ^^^^^^^ expected `Nat`, found `String`
+    │
```

Comment:
The order changing here doesn't seem right? Isn't this the primary error and the `Data/Nat.fun` part a hint?

Reply:
Yeah - this is an interesting design question. My long term plan was to order the labels in a single diagnostic first by the origin, and then by the line number, as opposed to prioritising the primary diagnostics. So the label messages should be readable no matter what order they are printed in. I realise this test doesn't demonstrate this well though. 

If folks want these diagnostics in order, then they could put them in separate diagnostics, say, with a follow-up 'note' diagnostic. Not sure though, willing to hear your thoughts on this!

Reply:
One potential solution of this is to make the `labels` section richer... allowing users to be more precise about whether they want to provide an ordered sequence, or an out of order set of labels, grouped by file id.

Reply:
Not sure what the best way is, but given a primary error and some help I would likely create the primary message first followed by the note and they would be created in conjuction. So I would expect that they were displayed in that way as well 🤷‍♂

Reply:
I guess what I'm preparing for is a world where we merge the labels (#100), and you get a diagnostic that reads like a bit of source code. Kind of like:

```

    ┌─ codespan/src/file.rs:102:34
    :
99  │  ╭──────── match line_index.compare(self.last_line_index()) {
100 │  │             Ordering::Less => Ok(self.line_starts()[line_index.to_usize()]),
    │  │                               --------------------------------------------- this is found to be of type `std::result::Result<index::ByteIndex, file::LineIndexOutOfBoundsError>`
101 │  │             Ordering::Equal => Ok(self.source_span().end()),
    │  │                                ---------------------------- this is found to be of type `std::result::Result<index::ByteIndex, file::LineIndexOutOfBoundsError>`
102 │  │             Ordering::Greater => LineIndexOutOfBoundsError {
    │ ╭│──────────────────────────────────^
103 │ ││                 given: line_index,
104 │ ││                 max: self.last_line_index(),
105 │ ││             },
    │ ╰│─────────────^ expected enum `std::result::Result`, found struct `file::LineIndexOutOfBoundsError`
106 │  │         }
    │  ╰─────────' `match` arms have incompatible types
    :
    = note: expected type `std::result::Result<index::ByteIndex, file::LineIndexOutOfBoundsError>`
               found type `file::LineIndexOutOfBoundsError`
```
Not sure how to support both label ordering mattering, and it not at the same time 🤔

Reply:
It is ok that they appear in a different order if the labels are merged, however in this case the diagnostics were from different files so they wouldn't be possible to merge. In that case the order is more important.